protocol/storage/client_test.go

```diff
@@ -106,7 +106,8 @@ func TestProposeDeal(t *testing.T) {
 
 		assert.Equal(t, storagedeal.Accepted, dealResponse.State)
 
-		retrievedDeal := testAPI.DealGet(dealResponse.ProposalCid)
+		retrievedDeal, err := testAPI.DealGet(dealResponse.ProposalCid)
+		require.NoError(t, err)
 
 		assert.Equal(t, retrievedDeal.Response, dealResponse)
```

Comment:
Please test the error return in the case of no deal found

Reply:
I'm not sure I understand your suggestion. Are you saying we should test ErrDealNotFound here, because I don't see how we could get that error when proposing a new deal?

Reply:
Not right at this line, but I was wanting test coverage of the tricky error logic that was incorrect in the first version of this PR. With closer inspection, I see the case I named is the expected one, already exercised. Is it possible to add to this test to exercise the case when the error *is* nil, because deal already in progress?